src/openrct2/cmdline/BenchSpriteSort.cpp

```diff
@@ -52,13 +52,13 @@ static void fixup_pointers(paint_session* s, size_t paint_session_entries, size_
         }
         for (size_t j = 0; j < quadrant_entries; j++)
         {
-            if (s[i].Quadrants[j] == (paint_struct*)quadrant_entries)
+            if (s[i].Quadrants[j] == reinterpret_cast<paint_struct*>(quadrant_entries))
             {
                 s[i].Quadrants[j] = nullptr;
             }
             else
             {
-                s[i].Quadrants[j] = &s[i].PaintStructs[(size_t)s[i].Quadrants[j]].basic;
+                s[i].Quadrants[j] = &s[i].PaintStructs[reinterpret_cast<size_t>(s[i].Quadrants[j])].basic;
```

Comment:
Think this can be a static_cast

Reply:
That actually gives a compile error

```
error: invalid static_cast from type ‘paint_struct*’ to type ‘size_t’ {aka ‘long unsigned int’}
   61 |                 s[i].Quadrants[j] = &s[i].PaintStructs[static_cast<size_t>(s[i].Quadrants[j])].basic;
```

It seems really odd to me that we just cast a big fat struct to an int, wouldn't it be better to just be using `image_id`? e.g.

`s[i].Quadrants[j] = &s[i].PaintStructs[s[i].Quadrants[j]->image_id].basic;`

I mean even here doing that weirdness of really just getting first struct member by casting:

https://github.com/OpenRCT2/OpenRCT2/blob/b628bba704b3b4dd0f5a1b548a4e718504259859/src/openrct2/cmdline/BenchSpriteSort.cpp#L49

Seems to me that also really should be 

`auto nextQuadrantPs = s[i].PaintStructs[j].basic.next_quadrant_ps->image_id;`

Reply:
Yeah this code is very odd thanks for looking into it properly. @janisozaur I can't understand what is trying to be achieved here. Has `s[i].Quadrants[j]` been loaded with offsets and this is converting them to pointers?

Reply:
Yes, that's correct. They're converted between offsets and pointers

Reply:
Where is this loading code? I tried to find it earlier but couldn't work out how it ends up with offsets.

Reply:
@duncanspumpkin Is it okay to merge this PR as-is, then? I imagine changing the offset/pointer conversion is a PR in its own right.